src/app/pages/communication/components/new-email/__tests__/new-email.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { ComponentFixture, async, TestBed } from '@angular/core/testing';
+import { IonicModule } from '@ionic/angular';
+import { By } from '@angular/platform-browser';
+import { TranslateModule, TranslateService, TranslateLoader } from '@ngx-translate/core';
+import { FormGroup, FormControl, ReactiveFormsModule } from '@angular/forms';
+import { HttpClient, HttpClientModule } from '@angular/common/http';
+import { configureTestSuite } from 'ng-bullet';
+import { createTranslateLoader } from '@app-module/app.module';
+import { NewEmailComponent } from '../new-email';
+
+describe('NewEmailComponent', () => {
+  let fixture: ComponentFixture<NewEmailComponent>;
+  let component: NewEmailComponent;
+  let translate: TranslateService;
+
+  configureTestSuite(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        NewEmailComponent,
+      ],
+      imports: [
+        IonicModule,
+        HttpClientModule,
+        ReactiveFormsModule,
+        TranslateModule.forRoot({
+          loader: {
+            provide: TranslateLoader,
+            useFactory: createTranslateLoader,
+            deps: [HttpClient],
+          },
+        }),
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(NewEmailComponent);
+    component = fixture.componentInstance;
+    translate = TestBed.get(TranslateService);
+    translate.setDefaultLang('en');
+    component.formGroup = new FormGroup({});
+    const radioCtrl = new FormControl('radioCtrl');
+    component.formGroup.addControl('radioCtrl', radioCtrl);
+    component.radioButtonControl = radioCtrl;
+    const newEmailCtrl = new FormControl('newEmailCtrl');
+    component.formGroup.addControl('newEmailCtrl', newEmailCtrl);
+    component.formControl = newEmailCtrl;
+    component.isNewEmailAddressChosen = true;
+  }));
+
+  describe('Class', () => {
+    describe('initialise', () => {
+      it('component should initialise', () => {
+        expect(component).toBeTruthy();
+      });
+    });
+  });
+
+  describe('DOM', () => {
+
+    describe('i18n', () => {
```

Comment:
tests appear different to DES3

Reply:
yes. the formatting in DES3 was awful. tests wernt linked to functions, there were tests that covered functionility covered in its components.